lib/node_modules/@stdlib/utils/jsonify-keys/lib/main.js

```diff
@@ -0,0 +1,75 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isFunction = require( '@stdlib/assert/is-function' );
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Attempts to parse a JSON-like string as JSON.
+*
+* @param {string} str - string to parse
+* @param {Function} [reviver] - transformation function
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} reviver must be a function
+* @returns {(*|Error)} parsed value or parse error
+*
+* @example
+* var obj = jsonifyKeys( '{ beep: "boop" }' );
+* // returns {'beep':'boop'}
+*/
+function jsonifyKeys( str, reviver ) {
+	var jsonStringified;
+	var withOperators;
+
+	if ( !isString( str ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', str ) );
+	}
+	if ( arguments.length > 1 ) {
+		if ( !isFunction( reviver ) ) {
+			throw new TypeError( format( 'invalid argument. Reviver argument must be a function. Value: `%s`.', reviver ) );
+		}
+	}
+	try {
+		// Handle keys prefixed with a `$` - useful for some cases like mongodb aggregator pipelines
+
+		// Hatches the keys by selecting the word next to a '{' or a ',' and adds quotes to it
+		withOperators = str.replace( /(\s*?{\s*?|\s*?,\s*?)(\$[a-zA-Z0-9_]+):/g, '$1"$2":' );
+
+		// Handle all generic keys
+
+		// Matches the keys by selecting the word next to a '{' or a ',' and adds quotes to it
+		jsonStringified = withOperators.replace( /(\s*?{\s*?|\s*?,\s*?)(['"])?([a-zA-Z0-9_]+)(['"])?:/g, '$1"$3":' );
```

Comment:
You should always minimize the amount of logic within a try/catch block, otherwise, you swallow non-operational errors, such as syntax errors.